core/modules/parsers/wikiparser/wikiparser.js

```diff
@@ -233,13 +243,21 @@ WikiParser.prototype.parseBlock = function(terminatorRegExpString) {
 	// Look for a block rule that applies at the current position
 	var nextMatch = this.findNextMatch(this.blockRules,this.pos);
 	if(nextMatch && nextMatch.matchIndex === this.pos) {
-		return nextMatch.rule.parse();
+		var start = this.pos;
+		var subTree = nextMatch.rule.parse();
+		// Set the start and end positions of the first and last blocks if they're not already set
+		if (subTree.length > 0) {
+			if (subTree[0].start === undefined) subTree[0].start = start;
+			if (subTree[subTree.length - 1].end === undefined) subTree[subTree.length - 1].end = this.pos;
+		}
+		$tw.utils.each(subTree, function (node) { node.rule = nextMatch.rule.name; });
+		return subTree;
 	}
 	// Treat it as a paragraph if we didn't find a block rule
 	var start = this.pos;
 	var children = this.parseInlineRun(terminatorRegExp);
 	var end = this.pos;
-	return [{type: "element", tag: "p", children: children, start: start, end: end }];
+	return [{type: "element", tag: "p", children: children, start: start, end: end, rule: null }];
```

Comment:
@Jermolene flagging for your review whether there might be a more appropriate value than null for the rule property here.

Reply:
My instinct would be to omit the "rule" property entirely in this case.

Reply:
This seem to be a bad idea now, because I need this info to restore \n\n in #8258